api/apps/api/src/modules/clone/infra/import/schedule-piece-import.handler.spec.ts

```diff
@@ -0,0 +1,178 @@
+import { API_EVENT_KINDS } from '@marxan/api-events';
+import {
+  ArchiveLocation,
+  ClonePiece,
+  ComponentId,
+  ResourceId,
+  ResourceKind,
+} from '@marxan/cloning/domain';
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { Logger } from '@nestjs/common';
+import { CqrsModule, EventBus, IEvent } from '@nestjs/cqrs';
+import { Test } from '@nestjs/testing';
+import { ApiEventsService } from '../../../api-events';
+import { MemoryImportRepository } from '../../import/adapters/memory-import.repository.adapter';
+import { ImportPieceFailed } from '../../import/application/import-piece-failed.event';
+import { ImportRepository } from '../../import/application/import.repository.port';
+import { Import, ImportComponent, ImportId } from '../../import/domain';
+import { importPieceQueueToken } from './import-queue.provider';
+import { SchedulePieceImport } from './schedule-piece-import.command';
+import { SchedulePieceImportHandler } from './schedule-piece-import.handler';
+
+function getDifferentComponentId(previous: ComponentId): ComponentId {
```

Comment:
This is not really needed (it is possible tho)

`A collision is possible but the total number of unique keys generated is so large that the possibility of a collision is almost zero. As per Wikipedia, the number of UUIDs generated to have atleast 1 collision is 2.71 quintillion. This is equivalent to generating around 1 billion UUIDs per second for about 85 years.`